modules/email-subscribe/email-subscribe.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/*
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ */
+class Email_Subscribe {
+
+	static $shortcode = 'jetpack-email-subscribe';
+
+	static $css_classname_prefix = 'jetpack-email-subscribe';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts_and_styles() {
+		wp_register_script( 'jetpack-email-subscribe', plugins_url( '/jetpack-email-subscribe.js', __FILE__ ), array( 'jquery' ) );
+		wp_register_style( 'jetpack-email-subscribe', plugins_url( '/jetpack-email-subscribe.css', __FILE__ ) );
+	}
+
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		$this->register_scripts_and_styles();
+		$this->register_shortcode();
+	}
+
+	function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
+	}
+
+
+	function parse_shortcode( $attrs, $content = false ) {
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'blog_id'     => $this->get_blog_id(),
+			'title'       => __( 'Join my email list' ),
```

Comment:
You will want to add the `jetpack` textdomain to all those strings.

Reply:
Done!